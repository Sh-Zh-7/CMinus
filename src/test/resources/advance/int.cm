int main()
{
    int i = 03;
    int j = -077;
    int k = 0123;

    int a = 0x6;
    int b = -0xabc;
    int c = 0X1a3Fe;
}