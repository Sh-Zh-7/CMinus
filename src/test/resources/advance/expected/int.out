Program(1)
  ExtDefList(1)
    ExtDef(1)
      Specifier(1)
        TYPE: int
      FunDec(1)
        ID: main
        LP
        RP
      CompSt(2)
        LC
        DefList(3)
          Def(3)
            Specifier(3)
              TYPE: int
            DecList(3)
              Dec(3)
                VarDec(3)
                  ID: i
                ASSIGNOP
                Exp(3)
                  INT: 3
            SEMI
          DefList(4)
            Def(4)
              Specifier(4)
                TYPE: int
              DecList(4)
                Dec(4)
                  VarDec(4)
                    ID: j
                  ASSIGNOP
                  Exp(4)
                    MINUS
                    Exp(4)
                      INT: 63
              SEMI
            DefList(5)
              Def(5)
                Specifier(5)
                  TYPE: int
                DecList(5)
                  Dec(5)
                    VarDec(5)
                      ID: k
                    ASSIGNOP
                    Exp(5)
                      INT: 83
                SEMI
              DefList(7)
                Def(7)
                  Specifier(7)
                    TYPE: int
                  DecList(7)
                    Dec(7)
                      VarDec(7)
                        ID: a
                      ASSIGNOP
                      Exp(7)
                        INT: 6
                  SEMI
                DefList(8)
                  Def(8)
                    Specifier(8)
                      TYPE: int
                    DecList(8)
                      Dec(8)
                        VarDec(8)
                          ID: b
                        ASSIGNOP
                        Exp(8)
                          MINUS
                          Exp(8)
                            INT: 2748
                    SEMI
                  DefList(9)
                    Def(9)
                      Specifier(9)
                        TYPE: int
                      DecList(9)
                        Dec(9)
                          VarDec(9)
                            ID: c
                          ASSIGNOP
                          Exp(9)
                            INT: 107518
                      SEMI
        RC

